src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -54,9 +58,15 @@ public void Dispose()
 
         public async Task StartAsync(CancellationToken cancellationToken)
         {
+            if (this.listernerStarted)
+            {
+                return;
```

Comment:
A listener that starts successfully should never be started again really. The various triggers aren't all consistent in their handling of this, however some of them (e.g. [ServiceBus](https://github.com/Azure/azure-webjobs-sdk/blob/a3153829592743b07239a9077318c8e2d4e78f62/src/Microsoft.Azure.WebJobs.ServiceBus/Listeners/ServiceBusListener.cs#L40)) throw in this case. The host won't call Start twice on a listener that successfully started. Your call.